mitodl/open-discussions
# Split posts and comments into separate Elasticsearch indices

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1337 

#### What's this PR do?
- Creates (and recreates) separate indices for posts and comments
- Creates an alias to allow searching both indices simultaneously
- Defines separate mappings for posts and comments
- Adds `channel_name` to both post and comment mappings

#### How should this be manually tested?
- Run `python manage.py recreate_index`
- After it's completed, posts should be found in a `discussions_local_post_default` index, comments in a `discussions_local_comment_default` index, and there should be an alias `discussions_local_all_default`.  The original `discussions_local_default` index from master branch will still exist unless you manually delete it or never created it in the first place.
- If you post the following to `http://localhost:9101/discussions_local_all_default/_search/`, you should get a list of posts back:
  ```javascript
  {
    "query": {
      "term": {
        "object_type": "post"
      }
    }
  }
  ```
- If you post the above but replace `post` with `comment`, you should get a list of comments back.

- Create a new post, then add some comments.  The new post should be added to the post index only, the comments should be added to the comment index only.  
- 'Remove' a post that has comments, then search for the records in Elasticsearch.  The comments should have `parent_post_removed: true`
